pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1542,7 +1542,13 @@ public CompletableFuture<Void> checkReplication() {
                 continue;
             }
             if (!replicators.containsKey(cluster)) {
-                futures.add(startReplicator(cluster));
+                futures.add(checkReplicationCluster(cluster).thenCompose(clusterExists -> {
```

Comment:
The method `checkReplicationCluster` is guarantee that the remote cluster is still in the collection `topicPolicies`. But the `line-1524` and `line-1540` also did this

Reply:
Yes, it's not related to the flaky test. I've reverted it and changed the PR description. The tests could still pass after that.
![image](https://user-images.githubusercontent.com/18204803/231054376-4422ff95-8d92-4d4b-8cae-ffe7b1d2a6da.png)